# Date: Wed, 2 Jun 2004 09:29:20 +1000
# From: Herbert Xu <herbert@gondor.apana.org.au>
# Changes: 
# Reported: http://bugs.debian.org/252187
# Security: No
# Upstream: Fixed

===== drivers/net/plip.c 1.20 vs edited =====
--- 1.20/drivers/net/plip.c	2004-03-04 05:52:24 +11:00
+++ edited/drivers/net/plip.c	2004-06-02 09:21:05 +10:00
@@ -1219,6 +1219,9 @@
 	struct net_local *nl = netdev_priv(dev);
 	struct plipconf *pc = (struct plipconf *) &rq->ifr_data;
 
+	if (cmd != SIOCDEVPLIP)
+		return -EOPNOTSUPP;
+
 	switch(pc->pcmd) {
 	case PLIP_GET_TIMEOUT:
 		pc->trigger = nl->trigger;
